### ISSUE TITLE
[Tooltip] a11y: Tooltip button should have optional aria-labelledby prop

### ISSUE THREAD

**Issue by amberfeinerman** [carbon-design-system/carbon#6615]
## Environment
**Operating system:** Mac OS

**Browser:** Chrome 84.0

**Assistive technology used to verify:** IBM Equal Access Accessibility Checker

## Detailed description

**What version of the Carbon Design System are you using?**
We are using Carbon through Cloud PAL version 1.190.0

**What WCAG 2.1 checkpoint does the issue violate?**
4.1.2 - Name, role, value

## Additional information
An [a11y issue](https://github.ibm.com/ibmcloud/security-compliance/issues/1317) was raised in my product team's repo regarding the Tooltip component. We are using it in a form label and it is causing an a11y error because:

> Form control with "button" has no associated label

The element it is referring to is `<div aria-expanded="false" tabindex="0" role="button" class="bx--tooltip__trigger">`. Because of the button role in the tooltip and us nesting it in a form, we need a unique id added to each instance of the tooltip OR an optional `aria-labelledby` prop.

![Screen Shot 2020-08-04 at 6 32 28 PM](https://user-images.githubusercontent.com/8129061/89355384-dec79080-d680-11ea-98cd-f6463c87be1c.png)

**Comment by tw15egan** [comment]
It seems like this error might be resolved by passing in a `triggerID` to the `Tooltip`, as this is used to add an `aria-labelledby` property

https://github.com/carbon-design-system/carbon/blob/2f2e17db587893ec53b2e03e758bfd9ba4a00433/packages/react/src/components/Tooltip/Tooltip.js#L502

**Comment by amberfeinerman** [comment]
Hi @tw15egan!

So here's what it looks like before I added the `triggerId` prop.

![Screen Shot 2020-08-06 at 3 52 40 PM](https://user-images.githubusercontent.com/8129061/89581630-edd34d80-d7fc-11ea-9d4a-7c9a73999356.png)

And here's what it looks like after:
![Screen Shot 2020-08-06 at 3 49 50 PM](https://user-images.githubusercontent.com/8129061/89581625-eca22080-d7fc-11ea-9896-565fe9e52a65.png)
![Screen Shot 2020-08-06 at 3 51 57 PM](https://user-images.githubusercontent.com/8129061/89581626-ed3ab700-d7fc-11ea-8663-151f0fbca342.png)

The issue is not resolved because we need the tooltip trigger to have the `aria-labelledby` attribute.

The code snippet you sent shows that the `aria-labelledby` prop is applied to the dialog, but the issue that is showing is that the `<div aria-expanded="false" tabindex="0" role="button" class="bx--tooltip__trigger">`  needs the `aria-labelledby` attribute because it has a role of button. (Screenshot below)

![Screen Shot 2020-08-06 at 3 56 01 PM](https://user-images.githubusercontent.com/8129061/89581961-72be6700-d7fd-11ea-82f2-d832920c1d74.png)

**Comment by tw15egan** [comment]
~Weird, I'm playing around with it now, but I'm seeing the `aria-labelledby` be applied if there you do **not** specify `triggerId` (auto-generated on our end).~


Noticing now that this is for a `Tooltip` that does _not_ contain visible text 🤦 ! Perhaps the `iconDescription` needs to be added?

**Comment by amberfeinerman** [comment]
@tw15egan That did the trick! Thanks!
